src/tests/unit/userService/generateToken.test.ts

```diff
@@ -0,0 +1,49 @@
+import 'dotenv/config';
+import UserService from '../../../services/userService';
+import { User } from '../../../models/userModel';
+import { IUser } from '../../../interfaces/user';
+import { saveUser } from '../../mocks/mockUser';
+import jwt, { JwtPayload } from 'jsonwebtoken';
+import { MATCH_JWT } from '../../constants';
+import { setupTests } from '../../setupTests';
+
+const userService = new UserService();
+const secret = process.env.TOKEN_KEY || 'secret';
+
+setupTests('generateToken', () => {
+  test('returns new set of tokens for a user', async () => {
+    const user = await saveUser();
+    const { accessToken, refreshToken } = await userService.generateToken(user);
+    expect(accessToken).toMatch(MATCH_JWT);
+    expect(refreshToken).toMatch(MATCH_JWT); //matches a JWT token
+  });
+
+  test('saves a refresh token to a user', async () => {
+    let user = await saveUser();
+    const { refreshToken } = await userService.generateToken(user);
+    user = <IUser>await User.findById(user.id);
```

Comment:
I wouldn't mutate `user` here, create a separate variable for the user found in db